src/test/java/com/yegor256/tojos/MnCsvTest.java

```diff
@@ -122,4 +128,65 @@ void putsKeyFirst(@Mktmp final Path temp) throws IOException {
             )
         );
     }
+
+    @Test
+    void handlesConcurrentModificationInDupMethod(@Mktmp final Path temp) {
+        final Collection<Map<String, String>> rows = new ArrayList<>(10_000);
+        final AtomicReference<Throwable> error = new AtomicReference<>();
+        final Path path = temp.resolve("key-test.json");
+        final Mono csv = new MnCsv(path);
+        final int threads = 2;
+        MatcherAssert.assertThat(
+            "works without concurrency conflicts",
+            new Together<>(
+                threads, thread -> {
+                switch (thread) {
+                    case 0:
+                        add(rows);
+                        break;
+                    case 1:
+                        dup(csv, rows, error);
+                        break;
+                    default:
+                        break;
+                }
+                return thread;
+            }).asList(), Matchers.hasSize(2)
+        );
+        if (error.get() != null) {
+            final Throwable err = error.get();
+            Assertions.fail(
+                String.format(
+                    "Test failed due to exception: %s",
+                    err.getClass().getSimpleName()
+                ),
+                err
+            );
+        }
+    }
+
+    private static void add(final Collection<Map<String, String>> rows) {
+        for (int idx = 0; idx < 10_000; idx += 1) {
+            rows.add(
+                new HashMap<>(
+                    Collections.singletonMap(
+                        String.format("key%d", idx),
+                        String.format("value%d", idx)
+                    )
+                )
+            );
+        }
+    }
+
+    private static void dup(final Mono csv,
```

Comment:
@VladMilash What does `dup` mean here? It seems, you just write something into `csv`. What do you duplicate here? Can we change the name of this method? Also it would be good to add a `javadoc` that would explain the purpose of this method. The same is relevant for the `add` method.